fix(enigma): one stream read per statement in client packet handlers

diff --git a/Source/gg2/Objects/Menus/Lobby Elements/LobbyController.events/Step.xml b/Source/gg2/Objects/Menus/Lobby Elements/LobbyController.events/Step.xml
index 5af37cf6..3f1651f6 100644
--- a/Source/gg2/Objects/Menus/Lobby Elements/LobbyController.events/Step.xml	
+++ b/Source/gg2/Objects/Menus/Lobby Elements/LobbyController.events/Step.xml	
@@ -49,7 +49,9 @@
                     server = instance_create(0,0,Server);
                     server.protocol = read_ubyte(lobbysocket);
                     server.port = read_ushort(lobbysocket);
-                    server.ip = string(read_ubyte(lobbysocket))+"."+string(read_ubyte(lobbysocket))+"."+string(read_ubyte(lobbysocket))+"."+string(read_ubyte(lobbysocket));
+                    // TODO(enigma): one read per statement; ENIGMA keeps C++'s unspecified argument order, g++ reads right-to-left
+                    server.ip = string(read_ubyte(lobbysocket));
+                    repeat (3) server.ip += "." + string(read_ubyte(lobbysocket));
                     read_string(lobbysocket, 2+16);
                     server.slots = read_ushort(lobbysocket);
                     server.players = read_ushort(lobbysocket);
diff --git a/Source/gg2/Scripts/Client/ClientBeginStep.gml b/Source/gg2/Scripts/Client/ClientBeginStep.gml
index b8396bc2..c4504749 100644
--- a/Source/gg2/Scripts/Client/ClientBeginStep.gml
+++ b/Source/gg2/Scripts/Client/ClientBeginStep.gml
@@ -1,5 +1,5 @@
 // receive and interpret the server's message(s)
-var i, playerObject, playerID, player, otherPlayerID, otherPlayer, sameVersion, buffer, usePlugins;
+var i, playerObject, playerID, player, otherPlayerID, otherPlayer, sameVersion, buffer, usePlugins, packetType, spawnpointId, spawnGroup, sentryX, sentryY;
 
 if(tcp_eof(global.serverSocket)) {
     if(gotServerHello)
@@ -36,7 +36,8 @@ if(downloadingMap)
 roomchange = false;
 do {
     if(tcp_receive(global.serverSocket,1)) {
-        switch(read_ubyte(global.serverSocket)) {
+        packetType = read_ubyte(global.serverSocket);
+        switch(packetType) {
         case HELLO:
             gotServerHello = true;
             global.joinedServerName = receivestring(global.serverSocket, 1);
@@ -313,7 +314,10 @@ do {
         case PLAYER_SPAWN:
             receiveCompleteMessage(global.serverSocket,3,global.tempBuffer);
             player = ds_list_find_value(global.players, read_ubyte(global.tempBuffer));
-            doEventSpawn(player, read_ubyte(global.tempBuffer), read_ubyte(global.tempBuffer));
+            // TODO(enigma): one read per statement; ENIGMA keeps C++'s unspecified argument order, g++ reads right-to-left
+            spawnpointId = read_ubyte(global.tempBuffer);
+            spawnGroup = read_ubyte(global.tempBuffer);
+            doEventSpawn(player, spawnpointId, spawnGroup);
             break;
               
         case CHAT_BUBBLE:
@@ -326,7 +330,10 @@ do {
         case BUILD_SENTRY:
             receiveCompleteMessage(global.serverSocket,6,global.tempBuffer);
             player = ds_list_find_value(global.players, read_ubyte(global.tempBuffer));
-            buildSentry(player, read_ushort(global.tempBuffer)/5, read_ushort(global.tempBuffer)/5, read_byte(global.tempBuffer));
+            // TODO(enigma): one read per statement; ENIGMA keeps C++'s unspecified argument order, g++ reads right-to-left
+            sentryX = read_ushort(global.tempBuffer)/5;
+            sentryY = read_ushort(global.tempBuffer)/5;
+            buildSentry(player, sentryX, sentryY, read_byte(global.tempBuffer));
             break;
               
         case DESTROY_SENTRY:
@@ -660,7 +667,7 @@ do {
             break;
 
         default:
-            promptRestartOrQuit("The Server sent unexpected data.");
+            promptRestartOrQuit("The Server sent unexpected data (packet " + string(packetType) + ").");
             exit;
         }
     } else {
